### PR TITLE
feat(arm): add CKV_AZURE_234 Ensure that Azure Defender for cloud is set to On for Resource Manager

### PR DESCRIPTION
**Pull request by userrut**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
Ensure that Azure Defender for cloud is set to On for Resource Manager, Checks if Azure Defender protection is enabled for cloud resources management.
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

I added a new security check named "Ensure that Azure Defender for cloud is set to On for Resource Manager" to our testing framework. The check examines the configuration of Azure Defender protection for the Resource Manager service in the cloud.
The motivation behind this change is to ensure that Azure Defender protection is properly configured and activated for Resource Manager by cloud administrators. By enforcing this check, we aim to enhance the security posture of cloud environments by ensuring that basic security configurations are in place according to organizational requirements.
This change requires the integration of the new security check into the existing testing framework. Additionally, it relies on the availability of Azure resources and configurations for testing purposes.

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/arm/checks/resource/AzureDefenderDisabledForResManager.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        id = "CKV_AZURE_234"
+        supported_resources = ("Microsoft.Security/pricings",)
+        categories = (CheckCategories.GENERAL_SECURITY,)
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf: dict[str, list[Any]]) -> CheckResult:
+        if "properties" in conf:
+            if "pricingTier" in conf["properties"]:
+                if str(conf["properties"]["pricingTier"]).lower() != "standard":
+                    if str(conf["properties"]["resourceType"]).lower() == "arm":
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/AzureDefenderDisabledForResManager.py:21]: no `resourceType` filed here -
https://learn.microsoft.com/en-us/azure/templates/microsoft.security/pricings?pivots=deployment-language-arm-template

Comment (userrut) [reply checkov/arm/checks/resource/AzureDefenderDisabledForResManager.py:21]: לשנות את ה-resourceType  ל-name של ה-resources?

File: checkov/arm/checks/resource/AzureDefenderDisabledForResManager.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        supported_resources = ("Microsoft.Security/pricings",)
+        categories = (CheckCategories.GENERAL_SECURITY,)
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf: dict[str, list[Any]]) -> CheckResult:
+        properties = conf.get('properties')
+        if properties and isinstance(properties, dict):
+            pricing_tier = properties.get("pricingTier")
+            if str(pricing_tier).lower() != "standard":
+                if str(conf.get("name", [None])).lower() == "arm":
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/AzureDefenderDisabledForResManager.py:22]: Why `name`?
the name can be everything.

Comment (userrut) [reply checkov/arm/checks/resource/AzureDefenderDisabledForResManager.py:22]: כיון ש-terrafom יש לי resourceType והוא חלק מהבדיקה?

**Comment by userrut** [comment]
כיון שב-terrafom יש בדיקה של עבור ה-Key Vaults ומשתמשים בפרמטר של-
resource_type
אז איך ניתן להגדיר במקום?


‫בתאריך יום ג׳, 28 במאי 2024 ב-20:11 מאת ‪ChanochShayner‬‏ <‪
***@***.***‬‏>:‬

**Pull request closed without merging**